integration/spark/src/main/scala/org/apache/carbondata/indexserver/DistributedRDDUtils.scala

```diff
@@ -108,7 +108,20 @@ object DistributedRDDUtils {
           val wrapper: IndexInputSplit = legacySegment
             .asInstanceOf[IndexInputSplitWrapper].getDistributable
           val executor = validExecutorIds(index % validExecutorIds.length)
-          wrapper.setLocations(Array("executor_" + executor))
+          tableToExecutorMapping.putIfAbsent(tableUniqueName,
```

Comment:
please add a detailed comment explaining this logic, so that next time if any issues, developer will be aware of the old issue fix

Reply:
done